### ISSUE TITLE
ConcurrentModificationException when running the tests

### ISSUE THREAD

**Issue by cbeust** [cbeust/testng#1331]
### TestNG Version

6.10

### Expected behavior

No exception.

### Actual behavior

Running the tests produces:

```
[TestNG] Reporter org.testng.reporters.JUnitReportReporter@60256d1b failed
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
	at java.util.ArrayList$Itr.next(ArrayList.java:851)
	at org.testng.reporters.JUnitReportReporter.getNextConfiguration(JUnitReportReporter.java:220)
	at org.testng.reporters.JUnitReportReporter.generateReport(JUnitReportReporter.java:105)
	at org.testng.TestNG.generateReports(TestNG.java:1175)
	at org.testng.TestNG.run(TestNG.java:1102)
```

**Comment by krmahadevan** [comment]
@cbeust - I am just curious.. Was this in TestNG 6.10 ?

The stack trace suggests that it originated from `getNextConfiguration()` method `line 220` but when I see the source code for `JUnitReportReporter` the method `getNextConfiguration()` is within the lines [250-266](https://github.com/cbeust/testng/blob/6.10/src/main/java/org/testng/reporters/JUnitReportReporter.java#L250-L266) 

[Line 220](https://github.com/cbeust/testng/blob/6.10/src/main/java/org/testng/reporters/JUnitReportReporter.java#L220) in this java class is an empty line..

So can you please help re-confirm on the version?

**Comment by cbeust** [comment]
Mmmh, indeed, let me try again when I'm back at that computer.

-- 
Cédric

**Comment by elbaloo** [comment]
@krmahadevan, @cbeust on 6.9.13 [line 220](https://github.com/cbeust/testng/blob/6.9.13/src/main/java/org/testng/reporters/JUnitReportReporter.java#L220) corresponds to a for loop iterating `List<ITestResult> confResults`, which is modified on every iteration. Seems like a visual error when the `removeAll(Collection<?> c)` call was left there on [6b7e679](https://github.com/cbeust/testng/commit/6b7e67953c0d374ea14ca4a9e77905b894e8aba2#diff-9adb221127c12f6e6928e96a7b014f6b).

**Comment by krmahadevan** [comment]
@elbaloo - That was fixed as part of [this](https://github.com/cbeust/testng/commit/ee78cc319f57dfb175755acecc400a7686a51339) commit and is part of the latest released version of TestNG (6.10)

**Comment by krmahadevan** [comment]
@cbeust - Can we close this issue ?
